tests/conftest.py

```diff
@@ -34,7 +34,17 @@ def xonsh_execer(monkeypatch):
 
 
 @pytest.yield_fixture
-def xonsh_builtins():
+def xonsh_events():
+    yield events
+    for name, oldevent in vars(events).items():
+        # Heavily based on transmogrification
+        klass = type(oldevent).__bases__[0]  # events.on_chdir -> <class on_chdir> -> <class Event>
+        newevent = events._mkevent(name, klass, klass.__doc__)
```

Comment:
Maybe this should pass in all bases, just in case, even if it only takes the docstring from the first base.